code/game/objects/items/mail.dm

```diff
@@ -146,17 +147,27 @@
 	var/list/goodies = generic_goodies
 
 	var/datum/job/this_job = recipient.assigned_role
+	var/is_mail_restricted = FALSE // certain roles and jobs (prisoner) do not receive generic gifts
+
 	if(this_job)
 		if(this_job.paycheck_department && department_colors[this_job.paycheck_department])
 			color = department_colors[this_job.paycheck_department]
+
 		var/list/job_goodies = this_job.get_mail_goodies()
+		is_mail_restricted = this_job.exclusive_mail_goodies
 		if(LAZYLEN(job_goodies))
-			// certain roles and jobs (prisoner) do not receive generic gifts.
-			if(this_job.exclusive_mail_goodies)
+			if(is_mail_restricted)
 				goodies = job_goodies
 			else
 				goodies += job_goodies
 
+	if(!is_mail_restricted)
+		// every quirk adds +5 to the final weighted list (regardless the number of items in quirk list)
+		for(var/datum/quirk/quirk as anything in body.quirks)
+			if(LAZYLEN(quirk.mail_goodies))
+				var/quirk_goodie = pick(quirk.mail_goodies)
+				goodies[quirk_goodie] = 5
```

Comment:
why? 5 is quite arbritrary

Reply:
Right now the weighted list is:

50 for generic items
50 for job items (most jobs)

I wanted to pick a number that was not too high or low so that stacking multiple quirks doesn't tilt the weighted list too much.  5% is a reasonable chance that comes from lots of D&D and roguelikes (1d20) to give low chances a decent possibility compared to how rare 1% is.

Reply:
could you include this explanation in your comment in code?

Reply:
Sure thing https://github.com/tgstation/tgstation/pull/70091/commits/c7bce552a2583947dc4c85d3d136934dd39d8a40